aviary/subsystems/propulsion/propeller_performance.py

```diff
@@ -2,8 +2,135 @@
+            )
+            self.add_output(
+                "helical_mach",
+                val=np.zeros(nn),
+                units="unitless",
+                desc="helical Mach number",
+            )
+            self.declare_partials("helical_mach", [
+                                  "tip_mach", "mach"], rows=arange, cols=arange)
+        elif out_type is OutMachType.MACH:
+            self.add_input(
+                "tip_mach",
+                val=np.zeros(nn),
+                units="unitless",
+                desc="tip Mach number at 3/4 of a blade",
+            )
+            self.add_input(
+                "helical_mach",
+                val=np.zeros(nn),
+                units="unitless",
+                desc="helical Mach number",
+            )
+            self.add_output(
+                "mach",
+                val=np.zeros(nn),
+                units="unitless",
+                desc="Mach number",
+            )
+            self.declare_partials("mach", [
+                                  "tip_mach", "helical_mach"], rows=arange, cols=arange)
+        elif out_type is OutMachType.TIP_MACH:
+            self.add_input(
+                "mach",
+                val=np.zeros(nn),
+                units="unitless",
+                desc="Mach number",
+            )
+            self.add_input(
+                "helical_mach",
+                val=np.zeros(nn),
+                units="unitless",
+                desc="helical Mach number",
+            )
+            self.add_output(
+                "tip_mach",
+                val=np.zeros(nn),
+                units="unitless",
+                desc="tip Mach number at 3/4 of a blade",
+            )
+            self.declare_partials("tip_mach", [
+                                  "mach", "helical_mach"], rows=arange, cols=arange)
+
+    def compute(self, inputs, outputs):
+        out_type = self.options["output_mach_type"]
+
+        if out_type is OutMachType.HELICAL_MACH:
+            mach = inputs["mach"]
+            tip_mach = inputs["tip_mach"]
+            outputs["helical_mach"] = np.sqrt(mach * mach + tip_mach * tip_mach)
+        elif out_type is OutMachType.MACH:
+            tip_mach = inputs["tip_mach"]
+            helical_mach = inputs["helical_mach"]
+            outputs["mach"] = np.sqrt(helical_mach * helical_mach - tip_mach * tip_mach)
+        elif out_type is OutMachType.TIP_MACH:
+            mach = inputs["mach"]
+            helical_mach = inputs["helical_mach"]
+            outputs["tip_mach"] = np.sqrt(helical_mach * helical_mach - mach * mach)
+
+    def compute_partials(self, inputs, J):
+        out_type = self.options["output_mach_type"]
+
+        if out_type is OutMachType.HELICAL_MACH:
+            mach = inputs["MACH"]
+            tip_mach = inputs["tip_mach"]
+            J["helical_mach", "MACH"] = -mach/np.sqrt(mach * mach + tip_mach * tip_mach)
+            J["helical_mach", "tip_mach"] = -tip_mach / \
```

Comment:
I think the signs here (and below) are flipped
```suggestion
            J["helical_mach", "MACH"] = mach/np.sqrt(mach * mach + tip_mach * tip_mach)
            J["helical_mach", "tip_mach"] = tip_mach / \
```

Reply:
done